usrsctplib/netinet/sctp_output.c

```diff
@@ -13901,7 +13901,7 @@ sctp_lower_sosend(struct socket *so,
 		}
 	}
 #if defined(__Userspace__)
-	if (inp->recv_callback) {
+	if (inp->recv_callback || SCTP_SO_IS_NBIO(so)) {
```

Comment:
This is not needed, since the check for `SCTP_SO_IS_NBIO(so)` is done in line 13908 for all platforms.